src/components/AddPub.js

```diff
@@ -0,0 +1,409 @@
+    this.props.changeManual(false);
+    this.props.changeError(null)
+  }
+
+  /**
+   * Called after a user recieves the information from the requested DOI and has
+   * had the oppurtunity to edit before submitting to be added to the database or 
+   * elected to submit manually.
+   * 
+   * If submitting manually, doi and title must be specified 
+   */
+  onContinue(e) {
+    e.preventDefault();
+    var data = this.props.doiInfo;
+    const dataObject = {
+      data: data,
+      token: this.token
+    };
+
+    if (
+      data["data"]["doi"] !== null &&
+      data["data"]["doi"] !== "" &&
+      data["data"]["title"] !== null &&
+      data["data"]["title"] !== ""
+    ) {
+      this.full = true;
+    } else {
+      this.full = false;
+      this.forceUpdate();
+    }
+
+    if (this.full) {
+      this.props.changeLoading(true);
+      console.log(data["data"])
+      this.props.postPubAction(dataObject);
+    }
+  }
+
+  /**
+   * Called when "Back to Home button is pressed"
+   */
+  onCancel(e) {
+    const newDoiInfo = {
+      data: {},
+      status: "empty",
+    };
+    this.props.changeDoiInfo(newDoiInfo);
+    this.props.history.push("/");
+    this.props.setFailure(false);
+    this.props.changeAddSuccess(false);
+    this.props.changeManual(false);
+  }
+
+  /**
+   * When a user elects to add a publication manually,
+   * the previous doi info (if any) is cleared
+   */
+  onManual(e) {
+    const newDoiInfo = {
+      data: {
+        title: null,
+        author: null,
+        publisher: null,
+        doi: null,
+        url: null,
+        abstract: null,
+        volume: null,
+        month: null,
+        year: null,
+      },
+      status: "empty",
+    };
+    this.props.setFailure(false);
+    this.data = [];
+    this.props.changeDoiInfo(newDoiInfo);
+    this.props.changeManual(true);
+  }
+
+  /**
+   * When a user elects to enter a DOI after previosly pressing 
+   * "enter manually". 
+   */
+  onDOI(e) {
+    const newDoiInfo = {
+      data: {},
+      status: "empty",
+    };
+    this.props.setFailure(false);
+    this.props.changeDoiInfo(newDoiInfo);
+    this.props.changeManual(false);
+    this.props.changeAddSuccess(false);
+  }
+
+  /**
+   * When a user logs out everything in props is reset and
+   * any class variables are reverted back to oringinal state
+   */
+  onLogout(e) {
+    const newDoiInfo = {
+      data: {},
+      status: "empty",
+    };
+    this.props.setFailure(false);
+    this.props.changeDoiInfo(newDoiInfo);
+    this.props.changeManual(false);
+    this.props.changeAddSuccess(false);
+    this.props.history.push("/");
+    this.props.keycloak["keycloak"].logout();
+  }
+
+  render() {
+    //main div is only displayed if user is authenticated by shib and in CIS group
+    if (this.props.keycloak["keycloak"]) {
+      if (this.props.keycloak["authenticated"]) {
+        if (this.props.keycloak["iscis"]) {
+          if (this.props.keycloak["profile"]) {
```

Comment:
I believe all the ifs are there so that we can show different messages on the screen to show which stage of loading it's in or what kind of error it is (like Loading Profile, Initializing, open a ticket)